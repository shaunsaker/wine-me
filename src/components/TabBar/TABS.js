const TABS = [
  {
    name: 'Home',
    iconName: 'home',
    sceneKey: '_home',
  },
  {
    name: 'Profile',
    iconName: 'person',
    sceneKey: '_profile',
  },
];

export default TABS;
