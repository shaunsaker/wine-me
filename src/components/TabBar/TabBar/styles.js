import { StyleSheet } from 'react-native';

import styleConstants from '../../../styleConstants';

const BORDER_BOTTOM_WIDTH = 2;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    backgroundColor: styleConstants.colors.primary,
    ...styleConstants.shadows.regular,
  },
  tabContainer: {
    flex: 1,
    alignItems: 'center',
    paddingVertical: styleConstants.dimensions.spacing.vertical / 2,
    paddingBottom: styleConstants.dimensions.spacing.vertical / 2 + BORDER_BOTTOM_WIDTH,
  },
  activeTabContainer: {},
  icon: {
    fontSize: styleConstants.fonts.sizes.icon,
    color: styleConstants.colors.white,
  },
  text: {
    ...styleConstants.fonts.types.small,
    color: styleConstants.colors.transWhite,
    ...styleConstants.fonts.mediumFontHelper,
  },
  activeText: {
    color: styleConstants.colors.white,
  },
  bottomBorder: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    borderBottomWidth: BORDER_BOTTOM_WIDTH,
    borderBottomColor: styleConstants.colors.white,
  },
});

export default styles;
