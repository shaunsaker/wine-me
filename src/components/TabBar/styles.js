import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    backgroundColor: styleConstants.colors.darkPrimary,
    ...styleConstants.shadows.regular,
  },
  tabContainer: {
    flex: 1,
    alignItems: 'center',
    paddingVertical: styleConstants.dimensions.spacing.vertical / 2,
  },
  activeTabContainer: {
    borderBottomWidth: 2,
    borderBottomColor: styleConstants.colors.white,
  },
  icon: {
    fontSize: styleConstants.fonts.sizes.icon,
    color: styleConstants.colors.white,
  },
  text: {
    ...styleConstants.fonts.types.small,
    color: styleConstants.colors.transWhite,
    ...styleConstants.fonts.mediumFontHelper,
  },
  activeText: {
    color: styleConstants.colors.white,
  },
});

export default styles;
