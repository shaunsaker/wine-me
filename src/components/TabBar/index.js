import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Actions } from 'react-native-router-flux';

import utils from '../../utils';

import TABS from './TABS';

import TabBar from './TabBar';

export class TabBarContainer extends React.Component {
  constructor(props) {
    super(props);

    this.onTabPress = this.onTabPress.bind(this);
    this.navigate = this.navigate.bind(this);

    this.state = {};
  }

  static propTypes = {
    scene: PropTypes.string,
  };

  static defaultProps = {};

  isCurrentScene(tab) {
    const { scene } = this.props;
    const isCurrentScene = tab.sceneKey === scene || tab.sceneKey === Actions.currentScene; // FIXME: scene is not always available on mount?

    return isCurrentScene;
  }

  onTabPress(tab) {
    this.navigate(tab.sceneKey);
  }

  navigate(page, props) {
    utils.app.navigate(page, props);
  }

  render() {
    const activeTab = TABS.filter((tab) => this.isCurrentScene(tab))[0];

    return <TabBar tabs={TABS} activeTab={activeTab} handleTabPress={this.onTabPress} />;
  }
}

function mapStateToProps(state) {
  return {
    scene: state.navigation.scene,
  };
}

export default connect(mapStateToProps)(TabBarContainer);
