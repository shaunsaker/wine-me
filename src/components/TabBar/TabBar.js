import React from 'react';
import PropTypes from 'prop-types';
import { View, Text } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';

import styles from './styles';

import Touchable from '../Touchable';

const propTypes = {
  tabs: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      iconName: PropTypes.string,
    }),
  ).isRequired,
  activeTab: PropTypes.shape({
    name: PropTypes.string,
    iconName: PropTypes.string,
  }).isRequired,
  handleTabPress: PropTypes.func,
};

const defaultProps = {};

const TabBar = ({ tabs, activeTab, handleTabPress }) => {
  return (
    <View style={styles.container}>
      {tabs.map((tab) => {
        const isActive = activeTab && activeTab.name === tab.name;

        return (
          <Touchable
            key={tab.name}
            onPress={() => handleTabPress(tab)}
            style={[styles.tabContainer, isActive && styles.activeTabContainer]}
          >
            <Icon name={tab.iconName} style={[styles.icon, isActive && styles.activeText]} />

            <Text style={[styles.text, isActive && styles.activeText]}>{tab.name}</Text>
          </Touchable>
        );
      })}
    </View>
  );
};

TabBar.propTypes = propTypes;
TabBar.defaultProps = defaultProps;

export default TabBar;
