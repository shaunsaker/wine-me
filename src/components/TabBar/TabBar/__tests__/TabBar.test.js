import React from 'react';
import renderer from 'react-test-renderer';

import TabBar from '..';
import TABS from '../../TABS';

describe('TabBar', () => {
  describe('renders', () => {
    it('renders', () => {
      const component = renderer.create(<TabBar tabs={TABS} />);

      expect(component).toMatchSnapshot();
    });

    it('renders the active state', () => {
      const component = renderer.create(<TabBar tabs={TABS} activeTab={TABS[0]} />);

      expect(component).toMatchSnapshot();
    });
  });
});
