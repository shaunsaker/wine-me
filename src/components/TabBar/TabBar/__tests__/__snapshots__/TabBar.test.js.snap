// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBar renders renders 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#48466D",
      "borderColor": null,
      "borderWidth": 0,
      "elevation": 6,
      "flexDirection": "row",
      "shadowColor": "#212121",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.33,
      "shadowRadius": 2,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "opacity": 1,
        "paddingBottom": 7,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 24,
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(255, 255, 255, 0.70)",
            "fontFamily": "Nunito-Regular",
            "fontSize": 15,
          },
          undefined,
        ]
      }
    >
      Home
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "opacity": 1,
        "paddingBottom": 7,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 24,
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(255, 255, 255, 0.70)",
            "fontFamily": "Nunito-Regular",
            "fontSize": 15,
          },
          undefined,
        ]
      }
    >
      Profile
    </Text>
  </View>
</View>
`;

exports[`TabBar renders renders the active state 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#48466D",
      "borderColor": null,
      "borderWidth": 0,
      "elevation": 6,
      "flexDirection": "row",
      "shadowColor": "#212121",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.33,
      "shadowRadius": 2,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "opacity": 1,
        "paddingBottom": 7,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 24,
            },
            Object {
              "color": "#FFFFFF",
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(255, 255, 255, 0.70)",
            "fontFamily": "Nunito-Regular",
            "fontSize": 15,
          },
          Object {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      Home
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#FFFFFF",
          "borderBottomWidth": 2,
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "opacity": 1,
        "paddingBottom": 7,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 24,
            },
            false,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(255, 255, 255, 0.70)",
            "fontFamily": "Nunito-Regular",
            "fontSize": 15,
          },
          false,
        ]
      }
    >
      Profile
    </Text>
  </View>
</View>
`;
